tests/framework/helpers/HtmlTest.php

```diff
@@ -100,6 +101,68 @@ public function testJsFile()
         $this->assertEquals("<!--[if (gte IE 9)|(!IE)]><!-->\n" . '<script src="http://example.com"></script>' . "\n<!--<![endif]-->", Html::jsFile('http://example.com', ['condition' => '(gte IE 9)|(!IE)']));
     }
 
+    public function testCsrfMetaTagsDisableCsrfValidation()
+    {
+        $this->assertEquals('', Html::csrfMetaTags());
+    }
+    
+    public function testCsrfMetaTagsEnableCsrfValidation()
+    {
+        $this->mockApplication([
+            'components' => [
+                'request' => [
+                    'class' => 'yii\web\Request',
+                    'enableCsrfValidation' => true,
+                    'cookieValidationKey' => 'key',
+                ],
+                'response' => [
+                    'class' => 'yii\web\Response',
+                ],
+            ],
+        ]);
+        $pattern = '<meta name="csrf-param" content="_csrf">%A<meta name="csrf-token" content="%s">';
+        $actual = Html::csrfMetaTags();
+        $this->assertStringMatchesFormat($pattern, $actual);
+    }
+    
+    public function testCsrfMetaTagsEnableCsrfValidationWithoutCookieValidationKey()
+    {
+        $request = $this->getMock('yii\\web\\Request');
+        $request->method('enableCsrfValidation')->willReturn(true);
+        Yii::$app->set('request', $request);
+        $pattern = '<meta name="csrf-param" content="_csrf">%A<meta name="csrf-token">';
+        $actual = Html::csrfMetaTags();
+        $this->assertStringMatchesFormat($pattern, $actual);
+    }
+
+    /**
+     * @dataProvider providerBeginFormSimulateViaPost
+     * 
+     * @param string $expected
+     * @param string $method
+     */
+    public function testBeginFormSimulateViaPost($expected, $method)
```

Comment:
Method `testBeginFormSimulateViaPost` and data `providerBeginFormSimulateViaPost` testing conition `framework/helpers/BaseHtml.php:336` neme for method i based on comment from `framework/helpers/BaseHtml.php:337`:

> simulate PUT, DELETE, etc. via POST